### DISCUSSION TITLE
Would this repository benefit from a dev container?

### DISCUSSION THREAD

**Discussion by corbob** [chocolatey-community/chocolatey-packages#2087] [Ideas]
I was in a discussion regarding migrating one of my packages to the vendor, and they noted they're all on Mac's. This reminded me of the docker container and got me to thinking: would this repository benefit from a dev container? Theoretically we could configure the container to have the AU module loaded, and anything else that is needed, then new comers could more easily work with the repository and verify their changes.

**Comment by pauby** [comment]
Do we need anything other than `chocolatey-au`? Do we need a container for that?

**Reply by corbob** [reply]
If I recall, testing the updates should result in packaged `nupkg`s which don't require a docker container per se, they do require having Chocolatey installed which means for macOS/Linux users either a VM or installing with mono.

Another potential benefit is that assuming all packages work on the `chocolatey/choco` docker container, the automated runs could be migrated and thus resolve any sort of "works on my machine" issues.

**Reply by JPRuskin** [reply]
One interesting thing I've tripped on recently has been that these scripts are all valid in Windows PowerShell, rather than pwsh.

That may torpedo the container bit for the reasons listed, unless we're willing to schedule and do some updates (definitely up for doing that, just a big/unknown change - as with certain other repositories).

As a counterpoint, though, if that weren't true I'd be very interested in having a dev container to ensure that I don't miss the "works on my machine" boat by having additional or incorrectly versioned modules or scripts installed (which also bit me last week), or missing modules (e.g. `Wormies-AU-Helpers`).

**Comment by AdmiringWorm** [comment]
While it could be helpful with a container being available, I don't think that is possible at the moment.

As James already mentioned, we need Windows PowerShell support (specifically v5), and can not rely on PowerShell Core at the moment (one of the big culprits of this that I can think of, is that some update scripts need to have IE available, which is possible through Windows PowerShell, but not through PowerShell core).

So, unless there is a container available right now with Windows PowerShell installed, it is not happening.

As for what is needed, at a minimum the container would need to have AU installed, my powershell extension Wormies-AU-Helpers, the chocolatey package chocolatey-core.extensions (currently not needed, but will in the future) and vscode with the editorconfig plugin installed (and enabled).
